lib/compat.php

```diff
@@ -24,3 +24,39 @@ function gutenberg_safe_style_css_column_flex_basis( $attr ) {
 	return $attr;
 }
 add_filter( 'safe_style_css', 'gutenberg_safe_style_css_column_flex_basis' );
+
+/**
+ * Adds an inline script to augment the default persistence method for data to
+ * use WordPress client user settings.
+ *
+ * @since 5.8.0
+ */
+function gutenberg_settings_cookie_persistence() {
+	// Ensure `utils` is a dependency, which makes available `userSettings`,
+	// `getUserSetting`, and `setUserSettings` window globals.
+	$data_script = wp_scripts()->query( 'wp-data', 'registered' );
+	if ( ! in_array( 'utils', $data_script->deps ) ) {
+		$data_script->deps[] = 'utils';
+	}
+
+	wp_add_inline_script(
+		'wp-data',
+		"
+( function() {
+	wp.data.use( wp.data.plugins.persistence, {
+		storage: {
+			getItem: function() {
+				return (
+					window.getUserSetting( 'dataState' ) ||
```

Comment:
How complex it would be to transform this to a package. To be clear, I think the current proposal is good enough, just wondering if we should take this opportunity to packagify this.

Reply:
Just saw the size limitations, it's unfortunate. What's the alternative you're thinking of? an endpoint?

Reply:
Yeah, I like how nice this solution is. I'd give it a try and see how it scales :)